tox.ini

```diff
@@ -1,20 +1,23 @@
 [tox]
 isolated_build = true
-envlist = black, pytest, mypy, pylint, bandit
+envlist = ruff_format, pytest, mypy, ruff_lint, bandit, py{3.8, 3.9, 3.10, 3.11}-sqla{14, 2}
 
```

Comment:
I like this but shouldn't the github actions be changed so it isn't doing the same thing

Reply:
Sort of, but kind of not. If you run tox on a PC that doesn't already have python 3.8 on it, it wouldn't be able to build the required environment for 3.8. You need to tell actions that you need all the version of python. However, I don't think what I have will actually work. I'm working on something to sort this out.